lib/engines/hcktest/platforms/schema.json

```diff
@@ -0,0 +1,130 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "name": {
+      "type": "string"
+    },
+    "kit": {
+      "type": "string"
+    },
+    "type": {
```

Comment:
The name `type` is confusing. we already have `machine_type` and `fw_type` so type of what?

We already call `setupmanager` the concept abstracting QemuHCK and PhysHCK, so it's better to follow the convention and name it `setup` also for consistency. We can name it `setup_type` for consistency with `machine_type` and `fw_type`, but there is `cpu`, which doesn't have `_type` suffix so I think either is fine. Personally I think `_type` suffix is verbose.